Add troubleshooting section (#601)
rancher/fleet: docs/troubleshooting.md

# Troubleshooting

> This is a collection of information, which help you to troubleshoot if fleet is not working

Reviewer: ```suggestion
This section contains commands and tips to troubleshoot Fleet.
```

## Check status of fleet controller

> You can check the status of the Fleet controller pods by running the below commands.

Reviewer: ```suggestion
You can check the status of the Fleet controller pods by running the commands below:
```

```bash
kubectl -n fleet-system logs -l app=fleet-controller
kubectl -n fleet-system get pods -l app=fleet-controller
```

```bash
NAME                                READY   STATUS    RESTARTS   AGE
fleet-controller-64f49d756b-n57wq   1/1     Running   0          3m21s
```

## Fleet fails with bad response code: 403

> If you see your GitJob complaining with this error

Reviewer: ```suggestion
If your GitJob returns the error below, the problem may be that Fleet cannot access the Helm repo you specified in your [`fleet.yaml`](./gitrepo-structure.md):
```

```
time="2021-11-04T09:21:24Z" level=fatal msg="bad response code: 403"
```

> This may indicate that fleet cannot access the helm repo you specified in your [`fleet.yaml`](./gitrepo-structure.md).

Reviewer: ```suggestion
Perform the following steps to assess:
```

> - check if your repo is accessible from your dev machine and you can actually download the helm chart

Reviewer: ```suggestion
- Check that your repo is accessible from your dev machine, and that you can download the Helm chart successfully
```

> - check if your credentials for the git repo are fine

Reviewer: ```suggestion
- Check that your credentials for the git repo are correct
```

## Helm chart repo: certificate signed by unknown authority

> If you see your GitJob complaining with this error

Reviewer: ```suggestion
If your GitJob returns the error below, you may have added the wrong certificate chain:
```

```
time="2021-11-11T05:55:08Z" level=fatal msg="Get \"https://helm.intra/virtual-helm/index.yaml\": x509: certificate signed by unknown authority" 
```

> You may have added the wrong certificate chain. Please verify with

Reviewer: ```suggestion
Please verify your certificate with the following command:
```
